kubernetes/cmsweb/docs/dmwm-documentation.md

```diff
@@ -0,0 +1,206 @@
+     [apervaiz@lxplus810 ~]$ kubectl -n kube-system get pods -l k8s-app=kube-dns
+      NAME                       READY   STATUS    RESTARTS        AGE
+      coredns-6d78487f9b-xjvcr   1/1     Running   0               21d
+      coredns-6d78487f9b-zgwpc   1/1     Running   0               21d
+
+     ```
+      Usually, restarting the deployment fixes the issue. Sometimes, you have to ensure that you are running the latest configuration. The configuration changes are made by       the CERN IT. The most common type of error is `FailedCreatePodSandBox        Failed to create pod sandbox: rpc error: code = Unknown desc = failed to set up sandbox container.`
+
+In summary, calico is responsible for managing networking and network security policies at the node level, ensuring connectivity and security for pods, while CoreDNS is the DNS server responsible for resolving domain names and enabling service discovery within the Kubernetes cluster. Together, they contribute to the effective functioning of a Kubernetes environment.
+
+
+#### Robot certificates/ Cron-proxy
+For cmsweb operations, we need to have a valid proxy on our backends. Originally, we used operator proxy and uploaded it to myproxy server. The
+cmsweb ProxySeed and ProxyRenew scripts were used.
+
+Now, we can obtain CERN service account and apply for a Robot certificate. Once received, we obtain robotcert.pem/robotkey.pem files and we need to
+register them in [LCG VOMS server](https://lcg-voms2.cern.ch:8443).  Then, we'll mount them in secrets volume in the k8s cluster and use them for acquiring proxies
+within k8s pods, see [proxy.sh](https://github.com/dmwm/CMSKubernetes/blob/master/docker/proxy/proxy.sh) file for example. It runs as a cronjob, named `cron-proxy.`
+
+#### Frontends
+
+When you want to add a new service to be hosted under CMSWeb, you have to take the following steps:
+  - The [deployment/frontend](https://github.com/dmwm/deployment/tree/master/frontend)
+repository contains SSL and NOSSL redirect rules for individual services or namespaces. They are labeled as app\_\<service-name\>\_ssl.conf and app\_\<service-name\>\_nossl.conf. You have to insert redirect rules there.
+  - Then, we need to insert rules in backends-k8s-prod.txt and backends-k8s-preprod.txt to let the frontends know where we would like our requests to be redirected. For K8s, you just need to supply a single `backends.txt` file with proper rules. This file is present in the [services_config/-/tree/cmsweb/frontend-ds](https://gitlab.cern.ch/cmsweb-k8s/services_config/-/tree/cmsweb/frontend-ds?ref_type=heads) branch for the production cluster, for preproduction cluster, it is present in the [services_config/-/tree/preprod/frontend-ds](https://gitlab.cern.ch/cmsweb-k8s/services_config/-/tree/preprod/frontend-ds?ref_type=heads), and for test cluster, it is present in [services_config/-/tree/test/frontend-ds](https://gitlab.cern.ch/cmsweb-k8s/services_config/-/tree/test/frontend-ds?ref_type=heads).
```

Comment:
Aroosha, do you think a table could be created for the environment vs branch hosting the frontends configuration? We might want to decide to place this content in a different place as well.